src/app/admin-app/component/admin-vlans/partition/partition.component.ts

```diff
@@ -0,0 +1,113 @@
+/*
+Copyright 2021 Carnegie Mellon University. All Rights Reserved. 
+ Released under a MIT (SEI)-style license. See LICENSE.md in the project root for license information.
+*/
+
+import {
+  Component,
+  OnInit,
+  ChangeDetectionStrategy,
+  Input,
+} from '@angular/core';
+import { MatSnackBar } from '@angular/material/snack-bar';
+import { Observable } from 'rxjs';
+import { Partition, Vlan } from 'src/app/generated/caster-api';
+import { ConfirmDialogService } from 'src/app/sei-cwd-common/confirm-dialog/service/confirm-dialog.service';
+import { PartitionQuery } from 'src/app/vlans/state/partition/partition.query';
+import { PartitionService } from 'src/app/vlans/state/partition/partition.service';
+import { VlanQuery } from 'src/app/vlans/state/vlan/vlan.query';
+import { VlanService } from 'src/app/vlans/state/vlan/vlan.service';
+
+@Component({
+  selector: 'cas-partition',
+  templateUrl: './partition.component.html',
+  styleUrls: ['./partition.component.scss'],
+  changeDetection: ChangeDetectionStrategy.OnPush,
+})
+export class PartitionComponent implements OnInit {
+  @Input() set poolId(poolId: string) {
```

Comment:
Thanks for the reminder that "@Input() set" is a thing.  This is very useful, but I keep forgetting about it!